### PR TITLE
Basicsetup unit testing

### PR DESCRIPTION
**Pull request by patelhr**
**Fixed Issues**

#21350

**Files Changed**

ServiceResponse.cs-Added this application class for error handling
setup.html-Added new validation field for port number and also added link for email settings.
setup.component.ts-Removed unused variable and Added response object for error handling.
setup.model.ts-Added ServiceResponse model for get response object from server side.
StringConstant.cs-Added error message constants.
BasicSetupController.cs-Changed comments.
BasicSetupRepositoryTest.cs-Added test cases for basic setup module.
bootstrap.cs-Added services and mocking for basic setup unit testing
DbUtility.cs-Added wrapper class for database operation.
FileUtility.cs-Added wrapper class for file operation.
BasicSetupRepository.cs-Moved code of Database and file operation to wrapper class for mocking.

**Checks**

- [x] Naming Conventions
- [x] Server side code comments (proper English, grammar and no spelling mistakes)
- [x] Client side code comments (proper English, grammar and no spelling mistakes)
- [x] Unused variables, methods, blank spaces and name spaces. 
- [x] Optimal Code and code formatting
- [x] Commit History is proper, no merge is done. 
- [x] Commit/pull request messages should be proper to justify your feature
- [x] All test cases are executed provided by tester.

**Known Issues**
N/A

**Comments**
N/A

### CODE REVIEW

File: Trappist/src/Promact.Trappist.Test/BasicSetup/BasicSetupRepositoryTest.cs [hunk 1]
```diff
@@ -1,12 +1,136 @@
...
     {
+        #region Private Variables
+        #region Dependencies
+        private readonly IBasicSetupRepository _basicSetupRepository;
+        private readonly Mock<IHostingEnvironment> _hostingEnvironmentMock;
+        private readonly Mock<IDbUtility> _sqlConnectionMock;
+        private readonly Mock<IFileUtility> _fileUtility;
+        private readonly Mock<IEmailService> _emailService;
+        #endregion
+        private BasicSetupModel basicSetup = new BasicSetupModel();
```
Comment (chintans) [inline Trappist/src/Promact.Trappist.Test/BasicSetup/BasicSetupRepositoryTest.cs]: MAJOR Make "basicSetup" "readonly". [rule](http://server12:9005/coding_rules#rule_key=csharpsquid%3AS2933)

**Comment by chintans** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

File: Trappist/src/Promact.Trappist.DomainModel/ApplicationClasses/BasicSetup/ServiceResponse.cs [hunk 2]
```diff
@@ -0,0 +1,8 @@
+﻿namespace Promact.Trappist.DomainModel.ApplicationClasses.BasicSetup
+{
+    public class ServiceResponse
+    {
+        public bool Response { get; set; }
```
Comment (iroshni) [inline Trappist/src/Promact.Trappist.DomainModel/ApplicationClasses/BasicSetup/ServiceResponse.cs]: IsSuccess is better name then Response.

File: Trappist/src/Promact.Trappist.Repository/BasicSetup/BasicSetupRepository.cs [hunk 3]
```diff
@@ -54,71 +54,48 @@ public class BasicSetupRepository : IBasicSetupRepository
                 Email = model.RegistrationFields.Email,
                 CreateDateTime = DateTime.UtcNow
             };
-            bool response = SaveSetupParameter(model);
+            var response = new ServiceResponse();
+            bool isSavedSetupParameter = SaveSetupParameter(model);
```
Comment (iroshni) [inline Trappist/src/Promact.Trappist.Repository/BasicSetup/BasicSetupRepository.cs]: isSetupParameterSavedSuccessfully is better to name.

File: Trappist/src/Promact.Trappist.Test/BasicSetup/BasicSetupRepositoryTest.cs [hunk 4]
```diff
@@ -1,12 +1,159 @@
...
+using Moq;
+using Promact.Trappist.Utility.DbUtil;
+using Promact.Trappist.Utility.FileUtil;
+using Promact.Trappist.Utility.EmailServices;
 
 namespace Promact.Trappist.Test.BasicSetup
 {
     [Collection("Register Dependency")]
-    public class BasicSetupRepositoryTest : BaseTest
+    public class BasicSetupRepositoryTest : Bootstrap
```
Comment (iroshni) [inline Trappist/src/Promact.Trappist.Test/BasicSetup/BasicSetupRepositoryTest.cs]: why you have removed BaseTest and kept BootStrap.

Comment (patelhr) [inline Trappist/src/Promact.Trappist.Test/BasicSetup/BasicSetupRepositoryTest.cs]: Ok.I will change it.

File: Trappist/src/Promact.Trappist.Repository/BasicSetup/BasicSetupRepository.cs [hunk 5]
```diff
@@ -136,7 +113,7 @@ private string GetConnectionString(SqlConnectionStringBuilder connectionString)
         /// </summary>
         /// <param name="model"></param>
         /// <returns>It return true or false</returns>
-        private bool SaveSetupParameter(BasicSetupModel model)
+        private bool isSetupParameterSavedSuccessfully(BasicSetupModel model)
```
Comment (chintans) [inline Trappist/src/Promact.Trappist.Repository/BasicSetup/BasicSetupRepository.cs]: MINOR Rename method "isSetupParameterSavedSuccessfully" to match camel case naming rules, consider using "IsSetupParameterSavedSuccessfully". [rule](http://server12:9005/coding_rules#rule_key=csharpsquid%3AS100)

**Comment by chintans** [comment]
SonarQube analysis reported 1 issue
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

File: Trappist/src/Promact.Trappist.Repository/BasicSetup/BasicSetupRepository.cs [hunk 6]
```diff
@@ -54,71 +54,52 @@ public class BasicSetupRepository : IBasicSetupRepository
                 Email = model.RegistrationFields.Email,
                 CreateDateTime = DateTime.UtcNow
             };
-            bool response = SaveSetupParameter(model);
+            var response = new ServiceResponse();
```
Comment (chintans) [inline Trappist/src/Promact.Trappist.Repository/BasicSetup/BasicSetupRepository.cs]: MAJOR Remove this useless assignment to local variable "response". [rule](http://server12:9005/coding_rules#rule_key=csharpsquid%3AS1854)

**Comment by chintans** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

**Comment by chintans** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [TrappistDbContext.cs#L90](https://github.com/Promact/trappist/blob/5b649220fc3c7700a0f544764460d486f6f75f2b/Trappist/src/Promact.Trappist.DomainModel/DbContext/TrappistDbContext.cs#L90): Use the overloading mechanism instead of the optional parameters. [rule](http://server12:9005/coding_rules#rule_key=csharpsquid%3AS2360)

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

**Comment by chintans** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [TrappistDbContext.cs#L90](https://github.com/Promact/trappist/blob/55fb79e8e0bc6f15caf9d28409f302f60b289c7a/Trappist/src/Promact.Trappist.DomainModel/DbContext/TrappistDbContext.cs#L90): Use the overloading mechanism instead of the optional parameters. [rule](http://server12:9005/coding_rules#rule_key=csharpsquid%3AS2360)

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

**Pull request merged**